Quickstart review (#6565)
Particular/docs.particular.net: tutorials/quickstart/tutorial-extending-the-system.md

---
title: "NServiceBus Quick Start: Extending the system"
reviewed: 2024-04-16
summary: "Part 3: Learn how easy it is to extend a distributed system by adding new functionality without affecting the other components of the system"
extensions:
- !!tutorial
  nextText: "NServiceBus Step-by-step tutorial"
  nextUrl: tutorials/nservicebus-step-by-step
---

> In [Part 1 of this tutorial](/tutorials/quickstart), you saw that publishing events using the [Publish-Subscribe pattern](/nservicebus/messaging/publish-subscribe/) reduces coupling and makes maintaining a system easier in the long run. Next, in [Part 2](/tutorials/quickstart/tutorial-reliability.md) you saw how to handle failures gracefully. Let's now look at how you can add an additional subscriber without needing to modify any existing code.

Reviewer: ```suggestion
In [Part 1 of this tutorial](/tutorials/quickstart), you saw that publishing events using the [Publish-Subscribe pattern](/nservicebus/messaging/publish-subscribe/) reduces coupling and makes maintaining a system easier in the long run. Then, in [Part 2](/tutorials/quickstart/tutorial-reliability.md) you saw how to handle failures gracefully. Now, let's look at how you can add an additional subscriber without needing to modify any existing code.
```

{{NOTE:

> If you didn't already, download the quick start project in the [previous lesson](/tutorials/quickstart), you can download it now:

Reviewer: Making this consistent with the other page:

```suggestion
If you didn't already download the Quickstart solution in the [previous lesson](/tutorials/quickstart), you can download it now:
```

downloadbutton
}}

As shown in the diagram, you will add a new messaging endpoint to the solution called **Shipping** that will subscribe to the `OrderPlaced` event.

![Completed Solution](after.svg "width=680")

NOTE: In this tutorial, you will use terminal commands like [`dotnet new`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new), [`dotnet add package`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-add-package), and [`dotnet add reference`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-add-reference), but you can do the same things using the graphical tools in your IDE if you prefer.

### Create a new endpoint
On this step, you will create the **Shipping** project and set up its dependencies.

> First let's make sure both browser windows and all console applications are closed, and in the terminal,  you're in the root of the project where the **RetailDemo.sln** file is located:

Reviewer: ```suggestion
First, make sure both browser windows and all console applications are closed. Then, in the terminal, navigate to the root of the project where the **RetailDemo.sln** file is located:
```

```shell
> cd tutorials-quickstart
```

> Next, you will create a new Console Application project named **Shipping** and add it to the solution:

Reviewer: ```suggestion
Next, create a new Console Application project named **Shipping** and add it to the solution:
```
